src/blockdata/script.rs

```diff
@@ -1245,5 +1262,22 @@ mod test {
 		let spending = Vec::from_hex("010000000001011f97548fbbe7a0db7588a66e18d803d0089315aa7d4cc28360b6ec50ef36718a0100000000ffffffff02df1776000000000017a9146c002a686959067f4866b8fb493ad7970290ab728757d29f0000000000220020701a8d401c84fb13e6baf169d59684e17abd9fa216c8cc5b9fc63d622ff8c58d04004730440220565d170eed95ff95027a69b313758450ba84a01224e1f7f130dda46e94d13f8602207bdd20e307f062594022f12ed5017bbf4a055a06aea91c10110a0e3bb23117fc014730440220647d2dc5b15f60bc37dc42618a370b2a1490293f9e5c8464f53ec4fe1dfe067302203598773895b4b16d37485cbe21b337f4e4b650739880098c592553add7dd4355016952210375e00eb72e29da82b89367947f29ef34afb75e8654f6ea368e0acdfd92976b7c2103a1b26313f430c4b15bb1fdce663207659d8cac749a0e53d70eff01874496feff2103c96d495bfdd5ba4145e3e046fee45e84a8a48ad05bd8dbb395c011a32cf9f88053ae00000000").unwrap();
 		spent.verify(0, 18393430, spending.as_slice()).unwrap();
 	}
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_script_serde_human_and_not() {
+        let script = Script::from(vec![0u8, 1u8, 2u8]);
+
+        // Serialize
+        let json = ::serde_json::to_string(&script).unwrap();
+        assert_eq!(json, "\"000102\"");
+        let bincode = ::bincode::serialize(&script).unwrap();
```

Comment:
[`serde-test`](https://docs.rs/serde_test) would allow you to test the specifics of human-readable / non-human-readable serialization without having to choose a particular format. At the same time, it also allows you to test the various overloads (`visit_string`, `visit_str`, etc).

Not testing these aspects caused a nasty bug that was [only fixed recently](https://github.com/rust-bitcoin/rust-secp256k1/pull/270/files) in `rust-secp256k1`.

I think our use of custom `Serialize`/`Deserialize` implementations warrants more thorough testing to make sure they are actually correct.

Reply:
I agree, but seems out of scope a bit. Would be nice to generally do this and not just in one place.

Reply:
A long time ago I tried using `serde-test`. I recall spending absolutely forever fighting its model of borrowed vs owned data and then finding that this didn't even match serde-json's model so that all my effort making serde-test work was useless because the encoding was still broken.

Reply:
Oh, that is interesting. Thanks for sharing!

I had a good experience with it when ironing out the serialization bugs in rust-secp which was precisely about owned vs borrowed data. That is why I mentioned it here.